tests/writer_test.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+"""Tests for the artifact definitions readers."""
+
+import io
+import os
+import unittest
+import json
+
+from artifacts import reader
+from artifacts import writer
+import artifacts.py2to3 as py2to3
+
+
+class YamlArtifactsWriterTest(unittest.TestCase):
+  """Class to test the YAML artifacts writer"""
+
+  def testYamlWriter(self):
+    """Tests the YamlArtifactsWriter FormatArtifacts method for loss during conversion."""
+    artifact_reader = reader.YamlArtifactsReader()
+    artifact_writer = writer.YamlArtifactsWriter()
+    test_file = os.path.join('test_data', 'definitions.yaml')
+
+    artifact_definitions = list(artifact_reader.ReadFile(test_file))
+    artifacts_yaml = artifact_writer.FormatArtifacts(artifact_definitions)
```

Comment:
your ascii conversion problem is with stringIO right?

Maybe just use an actual temp file and that way you can also exercise the actual file writer code

with tempfile.TemporaryFile() as converted:
  write yaml to converted
  do checks